docs/debugging-allocations.md

```diff
@@ -311,3 +311,206 @@ before: 10000, after: 20000
+free(): invalid pointer
+Aborted (core dumped)
+heaptrack stats:
+    allocations:              319347
+    leaked allocations:       107
+    temporary allocations:    68
+Heaptrack finished! Now run the following to investigate the data:
+
+  heaptrack --analyze "/tmp/.nio_alloc_counter_tests_GRusAy/heaptrack.test_1000_autoReadGetAndSet.84341.gz"
+```
+Then run it a second time for the feature branch:
+```
+ubuntu@ip-172-31-25-161 /t/.nio_alloc_counter_tests_ARusXy> heaptrack .build/x86_64-unknown-linux-gnu/release/test_1000_autoReadGetAndSet
+heaptrack output will be written to "/tmp/.nio_alloc_counter_tests_GRusAy/heaptrack.test_1000_autoReadGetAndSet.84372.gz"
+starting application, this might take some time...
+test_1000_autoReadGetAndSet.remaining_allocations: 0
+test_1000_autoReadGetAndSet.total_allocations: 0
+test_1000_autoReadGetAndSet.total_allocated_bytes: 0
+DEBUG: [["remaining_allocations": 0, "total_allocations": 0, "total_allocated_bytes": 0], ["total_allocated_bytes": 0, "total_allocations": 0, "remaining_allocations": 0], ["total_allocated_bytes": 0, "remaining_allocations": 0, "total_allocations": 0], ["total_allocations": 0, "total_allocated_bytes": 0, "remaining_allocations": 0], ["total_allocations": 0, "remaining_allocations": 0, "total_allocated_bytes": 0], ["remaining_allocations": 0, "total_allocations": 0, "total_allocated_bytes": 0], ["total_allocations": 0, "total_allocated_bytes": 0, "remaining_allocations": 0], ["remaining_allocations": 0, "total_allocated_bytes": 0, "total_allocations": 0], ["remaining_allocations": 0, "total_allocations": 0, "total_allocated_bytes": 0], ["remaining_allocations": 0, "total_allocated_bytes": 0, "total_allocations": 0]]
+free(): invalid pointer
+Aborted (core dumped)
+heaptrack stats:
+    allocations:              673989
+    leaked allocations:       117
+    temporary allocations:    341011
+Heaptrack finished! Now run the following to investigate the data:
+
+  heaptrack --analyze "/tmp/.nio_alloc_counter_tests_GRusAy/heaptrack.test_1000_autoReadGetAndSet.84372.gz"
+ubuntu@ip-172-31-25-161 /t/.nio_alloc_counter_tests_GRusAy> 
+```
+Here we could see that we had 673989 allocations in the feature branch version and 319347 in `main`, so clearly a regression.
+
+Finally, we can analyze the output as a diff from these runs using `heaptrack_print`:
+
+```
+heaptrack_print -T -d heaptrack.test_1000_autoReadGetAndSet.84341.gz heaptrack.test_1000_autoReadGetAndSet.84372.gz | swift demangle
+```
+`-T` gives us the temporary allocations (as it in this case was not a leak, but a transient alloaction - if you have leaks remove `-T`).
+
+The output can be quite long, but in this case as we look for transient allocations, scroll down to:
+
+```
+MOST TEMPORARY ALLOCATIONS
```

Comment:
I'd recommend truncating some of the output like in the other examples, it's quite a wall of text otherwise!

Reply:
Truncated and updated according to suggestions in b674ac0 now.